lib/tasks/journeys_report.rake

```diff
@@ -2,60 +2,98 @@ require 'csv'
 require 'json'
 require 'aws-sdk-s3'
 
+def get_last_run(bucket_name)
+  s3 = Aws::S3::Resource.new
+  if s3.bucket(bucket_name).object('bookmark').exists?
+    bookmark = s3.bucket(bucket_name).object('bookmark').get
+    last_run = JSON.parse(bookmark.body.read)['last_run']
+    last_run_date = DateTime.parse(last_run)
+  else
+    last_run_date = DateTime.strptime('0', '%s')
+  end
+  last_run_date
+end
+
+def update_bookmark(bucket_name)
+  s3 = Aws::S3::Resource.new
+  bookmark = { 'last_run': DateTime.now.utc.to_s }
+  s3.bucket(bucket_name).put_object(key: 'bookmark', body: JSON[bookmark])
+end
+
 namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
     timestamp = DateTime.now.utc.to_i
+    bucket_name = ENV['S3_REPORTING_BUCKET_NAME'] || raise('S3_REPORTING_BUCKET_NAME not defined')
+
+    if ENV['RAILS_ENV'].to_s == 'development' || ENV['RAILS_ENV'].to_s == ''
+      Aws.config.update(
+        endpoint: 'http://localhost:4572',
+        credentials: Aws::Credentials.new('fakeid', 'fakesecret'),
+        region: 'eu-west-2',
+        force_path_style: true,
+      )
+    end
+
+    last_report_date = get_last_run(bucket_name)
+    puts last_report_date
 
     Supplier.order(:key).each do |supplier|
       filename = "#{supplier.key}-#{timestamp}.json"
       puts "Generating #{filename}..."
       moves = []
       Move
           .where("ID in (select move_id from journeys WHERE supplier_id = '#{supplier.id}')")
+          .where("moves.date > '#{last_report_date}'")
```

Comment:
Watch out for three things: 
1. some pending moves don't have a date
2. move.date is only a date, not a datetime so what happens if you run the exporter twice on the same day (say 9am + 9pm) - you'll get duplicates
3. probably move.created_at is a better thing to segregate the date on, as all moves will have one and it will never change. Whereas move.date could be updated.

Reply:
How about updated_at.  My reasoning is that a move might have events or journeys added to it between reports.
We would want to send a updated version of the move on the following report, so that the HUB can update their records.
my assumption here is that updated_at would reflect events or journeys being added to the move, which might not necessarily be true.